### ISSUE TITLE
Option in settings to fold blocks to one line instead of two.

### ISSUE THREAD

**Issue by Detnator-Neo** [microsoft/vscode#121556]
When VSCode folds code the result is two lines that look something like this:

```
function something() { …
}
/* some comments …
*/
class somethingElse() { …
}
```

(note the ellipsis is my textual representation of the graphical three dots that appear in that location for folded code).

On the other hand, many IDEs fold (or have the option to fold) code up into one line that looks something like this:

```
function something() { … }
/* some comments … */
class somethingElse() { … 
}
```

Or...if you like screen shots: 
<img width="569" alt="New VSC folding options" src="https://user-images.githubusercontent.com/70407340/115123029-be2a3700-9f80-11eb-8463-2c321842b15b.png">

The one-line option obviously uses (up to) half less vertical space when everything is folded up.  Also, I find the one-line option much easier to read/follow/digest.

So I'd love it if there was a setting to choose which one we want. I imagine It would be a boolean something like "editor.foldToOneLine". Presumably its default value should be false, consistent with current behavior, but with a true option of course.

(If such a setting already exists, I'd love to know what it is -- can't find anything like it despite extensive searching).

**Comment by Detnator-Neo** [comment]
(Or if you want to be really brave you could make it an integer for the number of lines folded code folds up into: 2 or 1 respectively for the options above, but would allow the option for code to fold up to 3 or more lines if anyone ever wanted that.  Not something I'd like personally.  The boolean will do me fine. But if you're going there maybe it's minimal extra work to provide the integer option if anyone else wants that.)

**Comment by Detnator-Neo** [comment]
Don't know why I couldn't find it before I posted this (I did look).  It seems this is a duplicate of #3352.

For anyone else searching for this, go ahead and read all the comments in #3352, but ultimately, at least at the moment, it comes down to this (as summarized by @aeschli [here](https://github.com/microsoft/vscode/issues/3352#issuecomment-466334492) and [here](https://github.com/microsoft/vscode/issues/3352#issuecomment-513696419)):

This is actually two distinct issues: 

1. Folding up to one line instead of two (or hiding the last line as well as all the in between lines, instead of just the in between lines).
2. Displaying the closing bracket or tag or whatever else that currently otherwise shows on the second line, at the end of the folded up line with the graphical ellipsis in between.

Outcomes on this, at least to date: 
1. An extension exists that solves the first of those two issues (though it requires some configuration, but it works: [vscode-explicit-folding](https://github.com/zokugun/vscode-explicit-folding)).  
2. Solving this second issue is apparently not trivial and there are no plans to implement it (or at least there wasn't when @aeschli wrote that comment in 2019).

I suppose we should close this as a duplicate of #3352, but I'll let one of you devs/admins/whatever do that, since I'm kinda new here and not entirely sure how this works, and don't want to screw anything up.
